Source/FicsItNetworksLua/Private/FINLua/API/LuaDebugAPI.cpp

```diff
@@ -44,6 +44,31 @@ namespace FINLua {
 			 */)", traceback) { return 0; }
 		}
 
+		// TODO: how to properly document xpcall? Uncommenting this code crashes the game on loading a save.
+		//LuaModuleGlobalBareValue(R"(/**
+		// * @LuaGlobal		xpcall	fun(fn, ...): (ok: boolean, err: { message: any, trace: string }?)
+		// * @DisplayName		XPcall
+		// *
+		// * Ficsit networks ships with a slightly modifier version of `xpcall`. Like `pcall`, it accepts a function
+		// * and its arguments, calls it and checks whether an error has occurred.
+		// *
+		// * It returns a boolean indicating that the function call was successful.
+		// * If the call fails, it also returns a table with two fields: `message` is an error message
+		// * (or whatever was passed to an `error` call), and `trace` is a string with traceback.
+		// *
+		// * If the call was successful, our version of xpcall doesn't return its result.
+		// * To get it, pass in a closure that will set a local variable:
+		// *
+		// * [source,lua]
+		// * ---
+		// * local result
+		// * local ok, err = xpcall(function() result = doSomething() end)
+		// * ---
+		// */)", xpcall) {
+		//	// This function is here for documentation purposes only;
+		//	// The actual implementation is in LuaBaseModule.cpp
+		//}
+
```

Comment:
Due to the new Runtime, I will probably use the default xpcall soon again.

The reason for the custom xpcall was that very often the automatic yield occured within the handler function which is not yieldable. But with the new runtime there is a "last chance" mechanism that allows the code to continue executing until some other hard limit, and once it returns out of the handler function, the runtime will immidiately yield.